# Data Analysis

In [1]:
import pandas as pd
import numpy as np
import pathlib

PROJECT_ROOT = pathlib.Path().parent.parent
DATA_FOLDER = PROJECT_ROOT.joinpath('data')
KAGGLE_CLEANED = DATA_FOLDER.joinpath('kaggle_cleaned.csv')

kaggle_df = pd.read_csv(KAGGLE_CLEANED)

## Spliting data

In [2]:
from sklearn.model_selection import train_test_split

features = kaggle_df.columns[kaggle_df.columns != 'price']
X = kaggle_df[features]
y = kaggle_df['price']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=123)

print('Train size: ', len(X_train), 'Test size: ', len(y_test))

Train size:  433 Test size:  109


## Linear Regression

In [ ]:
""" Normal Split
This will split the data
"""

from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score

lr_model = LinearRegression()
lr_model.fit(X_train, y_train)

y_train_pred = lr_model.predict(X_train)

print("Results for linear regression on training data")
print(' Default settings')
print('Internal Parameters:')
print(' Bias is ', lr_model.intercept_)
print(' Coefficients', lr_model.coef_)
print(' Score', lr_model.score(X_train,y_train))
print()
print('Results for linear regression on train data')
print('MAE is ', mean_absolute_error(y_train,y_train_pred))
print('RMSE is ', np.sqrt(mean_squared_error(y_train, y_train_pred)))
print('RSE is ', mean_squared_error(y_train, y_train_pred))
print('R^2 ', r2_score(y_train, y_train_pred))

y_test_pred = lr_model.predict(X_test)
print()
print('Results for linear regression on test data')
print('MAE is ', mean_absolute_error(y_test,y_test_pred))
print('RMSE is ', np.sqrt(mean_squared_error(y_test, y_test_pred)))
print('RSE is ', mean_squared_error(y_test, y_test_pred))
print('R^2 ', r2_score(y_test, y_test_pred))

Results for linear regression on training data
 Default settings
Internal Parameters:
 Bias is  237206.20674365386
 Coefficients [ 3.07552160e+02  1.12465376e+05  1.02904399e+06  6.40034493e+05
  3.21626815e+05 -3.79562161e+05  3.79562161e+05]
 Score 0.5922244248979526

Results for linear regression on train data
MAE is  842098.3237588818
RMSE is  1136595.8387553697
RSE is  1291850100676.0225
R^2  0.5922244248979526

Results for linear regression on test data
MAE is  950042.4337808738
RMSE is  1227494.4747191283
RSE is  1506742685465.9885
R^2  0.5245574573425862


## Results
| Model | Type | RMSE | MAE | R^2 |
| --- | --- | --- | --- | --- |
| Linear Reg | Train | 1136595 | 842098 | 0.5922 |
| Linear Reg | Test | 1227494 | 950042 | 0.5246 |